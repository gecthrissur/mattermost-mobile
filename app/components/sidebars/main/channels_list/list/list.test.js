// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from '@mm-redux/constants/preferences';

import List from './list';

describe('ChannelsList List', () => {
    const baseProps = {
        onSelectChannel: jest.fn(),
        testID: 'main.sidebar.channels_list.list',
        canJoinPublicChannels: true,
        canCreatePrivateChannels: true,
        canCreatePublicChannels: true,
        favoriteChannelIds: [],
        unreadChannelIds: [],
        styles: {},
        theme: Preferences.THEMES.default,
        orderedChannelIds: [],
        isLandscape: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<List {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
