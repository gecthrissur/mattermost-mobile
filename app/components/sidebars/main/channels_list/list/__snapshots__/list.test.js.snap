// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelsList List should match snapshot 1`] = `
<View
  onLayout={[Function]}
  testID="main.sidebar.channels_list.list"
>
  <SectionList
    contentContainerStyle={
      Object {
        "paddingBottom": 44,
      }
    }
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardDismissMode="interactive"
    keyboardShouldPersistTaps="always"
    maxToRenderPerBatch={10}
    onEndReachedThreshold={2}
    onScrollBeginDrag={[Function]}
    onViewableItemsChanged={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={Array []}
    stickySectionHeadersEnabled={true}
    updateCellsBatchingPeriod={50}
    viewabilityConfig={
      Object {
        "itemVisiblePercentThreshold": 100,
        "waitForInteraction": true,
      }
    }
    windowSize={21}
  />
</View>
`;
